lib/aws-record/record/batch_write.rb

```diff
@@ -0,0 +1,83 @@
+# Copyright 2015-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+module Aws
+  module Record
+    class BatchWrite
+      attr_accessor(:client)
+
+      # @param [Aws::DynamoDB::Client] client the DynamoDB SDK client.
+      def initialize(client)
+        self.client = client
+      end
+
+      # Append a +PutItem+ operation to a batch write request.
+      #
+      # @param [Aws::Record] record a model class that includes {Aws::Record}.
+      def put(record)
+        table_name, params = record_put_params(record)
+        operations[table_name] ||= []
+        operations[table_name] << { put_request: params }
+      end
+
+      # Append a +DeleteItem+ operation to a batch write request.
+      #
+      # @param [Aws::Record] record a model class that includes {Aws::Record}.
+      def delete(record)
+        table_name, params = record_delete_params(record)
+        operations[table_name] ||= []
+        operations[table_name] << { delete_request: params }
+      end
+
+      # Perform a +batch_write_item+ request.
+      #
+      # @return [Aws::Record::BatchWrite] an instance that provides access to
+      #   unprocessed items and allows for retries.
+      def execute!
+        result = client.batch_write_item(request_items: operations)
+        @operations = result.unprocessed_items
+        self
+      end
+
+      # Indicates if any items exist that have not yet been processed.
+      #
+      # @return [Boolean] +true+ if +unprocessed_items+ are present, +false+
+      #   otherwise
+      def retryable?
```

Comment:
I'd lean towards calling this `complete?`

Reply:
Good call! I'll rename this.

Should we add an `incomplete?` as well?

Reply:
Maybe just keep things simple, it's easy enough to inverse the condition.